gui/dialogs/dialogsteadystate_t.cpp

```diff
@@ -0,0 +1,398 @@
+#include "../widgets/qlabelmousetracker.hpp"
+#include "../widgets/qvoxelrenderer.hpp"
+#include "catch_wrapper.hpp"
+#include "dialogdisplayoptions.hpp"
+#include "dialogsteadystate.hpp"
+#include "model_test_utils.hpp"
+#include "qt_test_utils.hpp"
+
+#include <QCheckBox>
+#include <QComboBox>
+#include <QLabel>
+#include <QPushButton>
+#include <QSlider>
+#include <catch2/catch_test_macros.hpp>
+#include <qtabwidget.h>
+#include <qtestsupport_core.h>
+#include <qwidget.h>
+#include <spdlog/spdlog.h>
+
+using namespace sme;
+using namespace sme::test;
+
+struct DialogSteadystateWidgets {
+  explicit DialogSteadystateWidgets(const DialogSteadystate *dialog) {
+    GET_DIALOG_WIDGET(QLabel, tolStepLabel);
+    GET_DIALOG_WIDGET(QLineEdit, convIntervalInput);
+    GET_DIALOG_WIDGET(QVoxelRenderer, valuesPlot3D);
+    GET_DIALOG_WIDGET(QLineEdit, timeoutInput);
+    GET_DIALOG_WIDGET(QLabel, zlabel);
+    GET_DIALOG_WIDGET(QLabel, timeoutLabel);
+    GET_DIALOG_WIDGET(QLabel, convergenceLabel);
+    GET_DIALOG_WIDGET(QFrame, line);
+    GET_DIALOG_WIDGET(QLabel, convIntervalLabel);
+    GET_DIALOG_WIDGET(QLineEdit, tolStepInput);
+    GET_DIALOG_WIDGET(QSlider, zaxis);
+    GET_DIALOG_WIDGET(QLabel, modeLabel);
+    GET_DIALOG_WIDGET(QComboBox, cmbPlotting);
+    GET_DIALOG_WIDGET(QLineEdit, toleranceInput);
+    GET_DIALOG_WIDGET(QComboBox, cmbConvergence);
+    GET_DIALOG_WIDGET(QLabel, valuesLabel);
+    GET_DIALOG_WIDGET(QLabelMouseTracker, valuesPlot);
+    GET_DIALOG_WIDGET(QLabel, toleranceLabel);
+    GET_DIALOG_WIDGET(QCustomPlot, errorPlot);
+    GET_DIALOG_WIDGET(QPushButton, btnStartStop);
+    GET_DIALOG_WIDGET(QPushButton, btnReset);
+    GET_DIALOG_WIDGET(QPushButton, btnDisplayOptions);
+    GET_DIALOG_WIDGET(QDialogButtonBox, buttonBox);
+  }
+
+  // Widgets for the dialog
+  QLabel *tolStepLabel;
+  QLineEdit *convIntervalInput;
+  QVoxelRenderer *valuesPlot3D;
+  QLineEdit *timeoutInput;
+  QLabel *zlabel;
+  QLabel *timeoutLabel;
+  QLabel *convergenceLabel;
+  QFrame *line;
+  QLabel *convIntervalLabel;
+  QLineEdit *tolStepInput;
+  QSlider *zaxis;
+  QLabel *modeLabel;
+  QComboBox *cmbPlotting;
+  QLineEdit *toleranceInput;
+  QComboBox *cmbConvergence;
+  QLabel *valuesLabel;
+  QLabelMouseTracker *valuesPlot;
+  QLabel *toleranceLabel;
+  QCustomPlot *errorPlot;
+  QPushButton *btnStartStop;
+  QPushButton *btnReset;
+  QPushButton *btnDisplayOptions;
+  QDialogButtonBox *buttonBox;
+};
+
+// helper for catching QMessageBox dialogs
+struct ModalChecker {
+  QTimer checker;
+  std::string messageContent;
+
+  void waitForConvergence(DialogSteadystate &dialog,
+                          const DialogSteadystateWidgets &widgets) {
+    messageContent = "";
+    // this didn't work with the ModalTimer we already have?. It always found
+    // the dialog itself and closed it before stuff was ready, and didn't latch
+    // onto the QMessageBox...?
```

Comment:
Did you try telling the modalwidgettimer to ignore the dialog itself?
 https://github.com/spatial-model-editor/spatial-model-editor/blob/d58351baa0ecb6a1bcb56ed2be1dd39e5d19aaf0/test/test_utils/qt_test_utils.hpp#L115